src/script.ts

```diff
@@ -150,27 +172,47 @@ const SUPPORTED_CODEMIRROR_NONGLOBAL_SITES = [
   { pattern: /https?:\/\/www\.codewars\.com(\/.*)?/, notebook: false },
   { pattern: /https:\/\/(.*\.)?github\.com(\/.*)?/, notebook: false },
 ];
-for (const pattern of SUPPORTED_CODEMIRROR_NONGLOBAL_SITES) {
-  if (pattern.pattern.test(window.location.href)) {
-    const codeMirrorState = new CodeMirrorState(extensionId, undefined, false);
-    const hook = codeMirrorState.editorHook();
-    setInterval(() => {
-      const docsByPosition = new Map<CodeMirror.Doc, number>();
-      for (const el of document.getElementsByClassName('CodeMirror')) {
-        const maybeCodeMirror = el as { CodeMirror?: CodeMirror.Editor };
-        if (maybeCodeMirror.CodeMirror === undefined) {
-          continue;
-        }
-        const editor = maybeCodeMirror.CodeMirror;
-        hook(editor);
-        if (pattern.notebook) {
-          docsByPosition.set(editor.getDoc(), (el as HTMLElement).getBoundingClientRect().top);
-        }
+
+const codeMirrorState = new CodeMirrorState(extensionId, undefined, false);
+const hook = codeMirrorState.editorHook();
+
+const addCodeMirror5Inject = () => {
+  setInterval(() => {
+    let notebook = false;
+    for (const pattern of SUPPORTED_CODEMIRROR_NONGLOBAL_SITES) {
+      if (pattern.pattern.test(window.location.href)) {
+        notebook = pattern.notebook;
+        break;
       }
-      if (pattern.notebook) {
-        const docs = [...docsByPosition.entries()].sort((a, b) => a[1] - b[1]).map(([doc]) => doc);
-        codeMirrorState.docs = docs;
+    }
+    const docsByPosition = new Map<CodeMirror.Doc, number>();
+    for (const el of document.getElementsByClassName('CodeMirror')) {
+      const maybeCodeMirror = el as { CodeMirror?: CodeMirror.Editor };
+      if (maybeCodeMirror.CodeMirror === undefined) {
+        continue;
+      }
+      const editor = maybeCodeMirror.CodeMirror;
+      hook(editor);
+      if (notebook) {
+        docsByPosition.set(editor.getDoc(), (el as HTMLElement).getBoundingClientRect().top);
       }
-    }, 100);
+    }
+    if (notebook) {
+      const docs = [...docsByPosition.entries()].sort((a, b) => a[1] - b[1]).map(([doc]) => doc);
+      codeMirrorState.docs = docs;
+    }
+  }, 100);
+};
+
+getWhiteList(extensionId).then((waitList) => {
+  for (const addr of waitList ?? []) {
+    const host = new RegExp(addr);
+    if (host.test(window.location.href)) {
+      // the url matches the whitelist
+      addMonacoInject();
+      addCodeMirrorInject();
```

Comment:
Is it possible that both `addCodeMirrorInject` and `addCodeMirror5Inject` will run on the same site?

Reply:
Yep, I tried on Code Share and it gave completion twice. Thanks for pointing the bug out.